BitFaster.Caching/Lru/CustomPolicy.cs

```diff
@@ -0,0 +1,305 @@
+    {
+        /// <summary>
+        /// Specify the inital time to live after an entry is created.
+        /// </summary>
+        TimeSpan GetExpireAfterCreate(K key, V value);
+
+        /// <summary>
+        /// Specify the time to live after an entry is read.
+        /// </summary>
+        TimeSpan GetExpireAfterRead(K key, V value);
+
+        /// <summary>
+        /// Specify the time to live after an entry is updated.
+        /// </summary>
+        TimeSpan GetExpireAfterUpdate(K key, V value);
+    }
+
+    /// <summary>
+    /// Defines a mechanism to determine the time to live for a cache item using function delegates.
+    /// </summary>
+    public readonly struct Expiry<K, V> : IExpiry<K, V>
+    {
+        private readonly Func<K, V, TimeSpan> expireAfterCreate;
+        private readonly Func<K, V, TimeSpan> expireAfterRead;
+        private readonly Func<K, V, TimeSpan> expireAfterUpdate;
+
+        /// <summary>
+        /// Initializes a new instance of the Expiry class.
+        /// </summary>
+        /// <param name="expireAfter">The delegate that computes the item time to live.</param>
+        public Expiry(Func<K, V, TimeSpan> expireAfter)
+        {
+            this.expireAfterCreate = expireAfter;
+            this.expireAfterRead = expireAfter;
+            this.expireAfterUpdate = expireAfter;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the Expiry class.
+        /// </summary>
+        /// <param name="expireAfterCreate">The delegate that computes the item time to live at creation.</param>
+        /// <param name="expireAfterRead">The delegate that computes the item time to live after a read operation.</param>
+        /// <param name="expireAfterUpdate">The delegate that computes the item time to live after an update operation.</param>
+        public Expiry(Func<K, V, TimeSpan> expireAfterCreate, Func<K, V, TimeSpan> expireAfterRead, Func<K, V, TimeSpan> expireAfterUpdate)
+        {
+            this.expireAfterCreate = expireAfterCreate;
+            this.expireAfterRead = expireAfterRead;
+            this.expireAfterUpdate = expireAfterUpdate;
+        }
+
+        ///<inheritdoc/>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public TimeSpan GetExpireAfterCreate(K key, V value)
+        {
+            return this.expireAfterCreate(key, value);
+        }
+
+        ///<inheritdoc/>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public TimeSpan GetExpireAfterRead(K key, V value)
+        {
+            return this.expireAfterRead(key, value);
+        }
+
+        ///<inheritdoc/>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public TimeSpan GetExpireAfterUpdate(K key, V value)
+        {
+            return this.expireAfterUpdate(key, value);
+        }
+    }
+
+#if NETCOREAPP3_0_OR_GREATER
+    internal readonly struct CustomExpiryPolicy<K, V> : IItemPolicy<K, V, LongTickCountLruItem<K, V>>

```

Comment:
There is currently no way for someone to write an equivalent because ConcurrentLruCore binds to this internal type.

Ideally, there should be a marker interface that is public.